fhir-java/src-gen/main/java/care/data4life/fhir/r4/model/MedicationStatement.java

```diff
@@ -0,0 +1,171 @@
+
+    // Reason for current status.
+    @Json(name = "statusReason")
+    @Nullable
+    public List<CodeableConcept> statusReason;
+
+    // Type of medication usage.
+    @Json(name = "category")
+    @Nullable
+    public CodeableConcept category;
+
+    // What medication was taken.
+    @Json(name = "medicationCodeableConcept")
+    @Nullable
+    public CodeableConcept medicationCodeableConcept;
+
+    // What medication was taken.
+    @Json(name = "medicationReference")
+    @Nullable
+    public Reference medicationReference;
+
+    // Who is/was taking  the medication.
+    @Json(name = "subject")
+    public Reference subject;
+
+    // Encounter / Episode associated with MedicationStatement.
+    @Json(name = "context")
+    @Nullable
+    public Reference context;
+
+    // The date/time or interval when the medication is/was/will be taken.
+    @Json(name = "effectiveDateTime")
+    @Nullable
+    public FhirDateTime effectiveDateTime;
+
+    // The date/time or interval when the medication is/was/will be taken.
+    @Json(name = "effectivePeriod")
+    @Nullable
+    public Period effectivePeriod;
+
+    // When the statement was asserted?.
+    @Json(name = "dateAsserted")
+    @Nullable
+    public FhirDateTime dateAsserted;
+
+    // Person or organization that provided the information about the taking of this medication.
+    @Json(name = "informationSource")
+    @Nullable
+    public Reference informationSource;
+
+    // Additional supporting information.
+    @Json(name = "derivedFrom")
+    @Nullable
+    public List<Reference> derivedFrom;
+
+    // Reason for why the medication is being/was taken.
+    @Json(name = "reasonCode")
+    @Nullable
+    public List<CodeableConcept> reasonCode;
+
+    // Condition or observation that supports why the medication is being/was taken.
+    @Json(name = "reasonReference")
+    @Nullable
+    public List<Reference> reasonReference;
+
+    // Further information about the statement.
+    @Json(name = "note")
+    @Nullable
+    public List<Annotation> note;
+
+    // Details of how medication is/was taken or should be taken.
+    @Json(name = "dosage")
+    @Nullable
+    public List<Dosage> dosage;
+
+
+    /**
+     * Constructor for all required properties.
+     *
+     * @param status     CodeSystemMedicationStatusCodes
+     * @param medication as one of CodeableConcept, Reference*
+     * @param subject    Reference
+     */
+    public MedicationStatement(CodeSystemMedicationStatusCodes status, Object medication, Reference subject) {
+        this.status = status;
+        if (medication instanceof CodeableConcept) {
+            this.medicationCodeableConcept = (CodeableConcept) medication;
+        } else if (medication instanceof Reference) {
+            this.medicationReference = (Reference) medication;
+        } else {
+            //FIXME Type: (of: medication)) for property (medication) is invalid, ignoring")
+        }
```

Comment:
Is this branch necessary? Also I am not sure, what the comment should exactly tell me - so it might be worth to make it more clear.

Reply:
A nitpick: @wmontwe